src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs

```diff
@@ -48,7 +48,15 @@ private static CSharpCompilation CreateCompilation(string source, IEnumerable<Me
 
         internal static CSharpCompilation CreateRuntimeAsyncCompilation(CSharpTestSource source, CSharpCompilationOptions options = null, CSharpParseOptions parseOptions = null)
         {
-            return CreateCompilation(source, options: options, parseOptions: parseOptions ?? WithRuntimeAsync(TestOptions.RegularPreview), targetFramework: TargetFramework.Net100);
+            parseOptions ??= WithRuntimeAsync(TestOptions.RegularPreview);
+            var syntaxTrees = source.GetSyntaxTrees(parseOptions, sourceFileName: "");
+            if (options == null)
+            {
+                options = CheckForTopLevelStatements(syntaxTrees);
+                options = options.WithSpecificDiagnosticOptions("SYSLIB5007", ReportDiagnostic.Suppress);
```

Comment:
Is this how these helpers are shipping in net10? It's basically opting in to an experiment?

Reply:
Yep, that's the plan.